compiler/Eta/Types/Coercion.hs

```diff
@@ -361,7 +361,7 @@ Note [Roles]
 ~~~~~~~~~~~~
 Roles are a solution to the GeneralizedNewtypeDeriving problem, articulated
 in Trac #1496. The full story is in docs/core-spec/core-spec.pdf. Also, see
-http://ghc.haskell.org/trac/ghc/wiki/RolesImplementation
+https://gitlab.haskell.org/ghc/ghc/-/wikis/roles-implementation
```

Comment:
Yes, I know that the text still says it's Trac and that it in fact is now in gitlab. I consciously chose not to update the text. The item was indeed articulated in `Trac #1496`, it just so happens that the resource now lives elsewhere.

If someone wants to rewrite the prose, they can.